services/IncrementService.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Craft;
+
+/**
+ * Increment Field Type.
+ *
+ * Number field with automatic incrementing.
+ *
+ * @author    Bob Olde Hampsink <b.oldehampsink@itmundi.nl>
+ * @copyright Copyright (c) 2015, Bob Olde Hampsink
+ * @license   MIT
+ *
+ * @link      http://github.com/boboldehampsink
+ */
+class IncrementService extends BaseApplicationComponent
+{
+
+    /**
+     * @param $fieldHandle
+     * @return mixed
+     */
+    public function getNewIncrement($fieldHandle, $minValue)
+    {
+        $maxValue = craft()->db->createCommand()->select('MAX(`field_' . $fieldHandle . '`)')->from('content')->queryScalar();
+
+        return $minValue > $maxValue ? $minValue : ($maxValue + 1);
+    }
+
+    /**
+     * @param EntryModel $entry
+     */
+    public function recalculateIncrements(EntryModel $entry)
```

Comment:
This could be done in the fieldtype too, so you won't have to loop through all fields